src/main/java/org/jenkinsci/plugins/codesonar/services/CodeSonarCacheService.java

```diff
@@ -193,11 +214,35 @@ public CodeSonarAlertFrequencies getAlerts(URI baseHubUri, long analysisId) thro
         return frequencies;
     }
     
+    public CodeSonarWarningCount getWarningCountAbsoluteWithScoreAboveThreshold(URI baseHubUri, long analysisId, int threshold) throws IOException {
+        LOGGER.log(Level.INFO, "getWarningCountAbsoluteWithScoreAboveThreshold threshold={0}", threshold);
+        CodeSonarAnalysisData analysisData = getAnalysisDataFromCache(baseHubUri, analysisId);
+        if(analysisData.getWarningCountAbsoluteWithScoreAboveThreshold() == null) {
+            LOGGER.log(Level.INFO, "Warnings with score above threshold not already set, loading from corresponding service");
+            analysisData.setWarningCountAbsoluteWithScoreAboveThreshold(getWarningsService().getNumberOfWarningsWithScoreAboveThreshold(baseHubUri, analysisId, threshold));
+        }
+        CodeSonarWarningCount warningCountAbsoluteWithScoreAboveThreshold = analysisData.getWarningCountAbsoluteWithScoreAboveThreshold();
+        LOGGER.log(Level.INFO, "CodeSonarWarningCount new instance {0}", warningCountAbsoluteWithScoreAboveThreshold);
+        return warningCountAbsoluteWithScoreAboveThreshold;
+    }
+    
+    public CodeSonarWarningCount getWarningCountIncreaseWithScoreAboveThreshold(URI baseHubUri, long analysisId, int threshold) throws IOException {
+        LOGGER.log(Level.INFO, "getWarningCountIncreaseWithScoreAboveThreshold threshold={0}", threshold);
+        CodeSonarAnalysisData analysisData = getAnalysisDataFromCache(baseHubUri, analysisId);
+        if(analysisData.getWarningCountIncreaseWithScoreAboveThreshold() == null) {
+            LOGGER.log(Level.INFO, "Warnings with score above threshold not already set, loading from corresponding service");
+            analysisData.setWarningCountIncreaseWithScoreAboveThreshold(getWarningsService().getNumberOfWarningsWithScoreAboveThreshold(baseHubUri, analysisId, threshold));
+        }
+        CodeSonarWarningCount warningCountIncreaseWithScoreAboveThreshold = analysisData.getWarningCountIncreaseWithScoreAboveThreshold();
+        LOGGER.log(Level.INFO, "CodeSonarWarningCount new instance {0}", warningCountIncreaseWithScoreAboveThreshold);
+        return warningCountIncreaseWithScoreAboveThreshold;
+    } 
+    
     public CodeSonarAnalysisData getCodeSonarAnalysisData(URI baseHubUri, long analysisId, String visibilityFilterAll, String visibilityFilterNew) throws IOException {
         LOGGER.log(Level.INFO, "getCodeSonarAnalysisData: baseHubUri = {0}, analysisId = {1,number,0}, visibilityFilterAll = {2}, visibilityFilterNew = {3}", new Object[] {baseHubUri, analysisId, visibilityFilterAll, visibilityFilterNew});
         getNumberOfActiveWarnings(baseHubUri, analysisId, visibilityFilterAll);
         getNumberOfNewWarnings(baseHubUri, analysisId, visibilityFilterNew);
-        getAnalysisActiveWarnings(baseHubUri, analysisId, visibilityFilterAll);
+//        getAnalysisActiveWarnings(baseHubUri, analysisId, visibilityFilterAll);

```

Comment:
Commented these two lines in order to temporarily avoid calling the old XML API for analysis.
It should be done the same for `getProcedures()` and `getMetrics()`.

Reply:
This marks a big TODO.  We must address backwards compatibility with older hubs.